nixos/modules/services/web-apps/turtl-server.nix

```diff
@@ -0,0 +1,385 @@
+          description = ''
+            Path where the plugins are installed.
+          '';
+        };
+
+        analytics= mkOption {
+          type = types.bool;
+          default = false;
+          description = ''
+            Enable analytics plugin.
+          '';
+        };
+
+        email= mkOption {
+          type = types.bool;
+          default = false;
+          description = ''
+            Enable email plugin.
+          '';
+        };
+
+        sync= mkOption {
+          type = types.bool;
+          default = false;
+          description = ''
+            Enable sync plugin.
+          '';
+        };
+      };
+
+      uploads = {
+        local = mkOption {
+          type = types.nullOr types.str;
+          default = "/var/www/turtl/server/public/uploads";
+          example = "/var/www/turtl/server/public/uploads";
+          description = ''
+            If set to a path, files will be uploaded to the local filesystem instead of S3. otherwise, set to null.
+          '';
+        };
+
+        localProxy= mkOption {
+          type = types.bool;
+          default = true;
+          description = ''
+            If true, downloading local files will be proxied through the turtl server.  This avoids needing to set up any CORS config in your favorite webserver, but may slightly affect performance on high-demand servers.
+          '';
+        };
+
+        url = mkOption {
+          type = types.str;
+          default = "http://api.turtl.dev/uploads";
+          example = "http://api.turtl.dev/uploads";
+          description = ''
+            If local_proxy is false, this is should be the url path the uploaded files are publicly available on.
+          '';
+        };
+      };
+
+      s3 = {
+        token = mkOption {
+          type = types.str;
+          default = "IHADAPETSNAKEBUTHEDIEDNOOOOO";
+          description = ''
+            S3 token.
+          '';
+        };
+
+        secret = mkOption {
+          type = types.str;
+          default = "";
+          description = ''
+            S3 secret.
+          '';
+        };
+
+        bucket = mkOption {
+          type = types.str;
+          default = "";
+          description = ''
+            S3 bucket.
+          '';
+        };
+
+        endpoint = mkOption {
+          type = types.str;
+          default = "https://s3.amazonaws.com";
+          description = ''
+            S3 endpoint.
+          '';
+        };
+      };
+
+    };
+  };
+
+  config = mkMerge [
+    (mkIf cfg.enable {
+      users.users = optionalAttrs (cfg.user == "turtl") (singleton {
+        name = "turtl";
+        group = cfg.group;
+        uid = config.ids.uids.turtl;
+        home = cfg.statePath;
+      });
+
+      users.groups = optionalAttrs (cfg.group == "turtl") (singleton {
+        name = "turtl";
+        gid = config.ids.gids.turtl;
+      });
+
+      services.postgresql.enable = cfg.localDatabaseCreate;
```

Comment:
Use `services.postgresql.enable = mkIf cfg.localDatabaseCreate true` instead. Otherwise this would try to turn of postgresql if the setting is false (and throw an error because it can be defined multiple times).